### ISSUE TITLE
Heatblock mounting screw type

### ISSUE THREAD

**Issue by LimesKey** [bassamanator/everything-sovol-sv06#19]
I am in need of some longer heatblock mounting screws as the one that comes stock with the printer is a bit too short for my new heatblock as in issue #16. The issue I'm running into is my digital calipers are not the greatest and are displaying different values for the diameter of the screw, anywhere from 2.2 - 2.5mm in diameter. Is someone else with metal and high resolution digital calipers able to find the correct diameter for me?
![image](https://user-images.githubusercontent.com/85136735/227695726-b2d7193b-7185-4782-8fc9-e49e0e60af4b.png)
I know there is a diagram on the specs of the heatblock but those are likely to be incorrect.

**Comment by bassamanator** [comment]
Not quite sure what you're asking exactly, however, if you asking about the bolts that mount the heatblock to the extruder, they are M2.5 x 10mm.

**Comment by LimesKey** [comment]
Yeah that is exactly what I am asking, are you 100% sure they are M2.5; the stocks one that come with extruder assembly. They could be slightly different than M2.5 as apparently there are many variations of it.

**Comment by bassamanator** [comment]
@LimesKey I'm as sure as I can be. I have M2.5 nuts that are definitely M2.5, and I used them as a gauge.

**Comment by LimesKey** [comment]
Thank you so much, I will update my pull request to add this information. All I will need then is just longer M2.5 screws

**Comment by bassamanator** [comment]
I picked up some SHCS and countersunk M2.5 screws on the aliexpress sale going on right now just to be able to deal with this heatblock.

I haven't forgotten about your PR btw.

**Comment by LimesKey** [comment]
Can you send me the link to them?

**Comment by bassamanator** [comment]
https://www.aliexpress.com/item/1005002187969166.html?spm=a2g0o.order_detail.order_detail_item.13.d2fcf19cmEm72Y
https://www.aliexpress.com/item/10000139502321.html?spm=a2g0o.order_detail.order_detail_item.4.d2fcf19cmEm72Y

I got the stainless steel and picked up a few sizes so that I can sand them down if need be. 16mm, 20mm, and 25mm just in case.

**Comment by LimesKey** [comment]
Thank you, that's great. The screws I picked up are a little worse as they are not SHCS. My printer hasn't been working for 4+ months now as every part I've ordered either wasn't the correct one or another issue emerged. Are you on the same journey? 

The main issue that caused this is that I got a big blob on the hotend and it destoryed the thermistor wires and it's been a struggle since then.

**Comment by bassamanator** [comment]
I've been lucky, I only have a warped bed. Last night, however, I discovered that it's a sensor issue: I managed to manually edit the bed mesh and the first layer goes on great now. Still have some work to do, but a truly bizarre problem. Bizarre because I've done the silicone mod, and have manually adjusted the bed level be be essentially perfect, yet the first layer goes on with gaps on the right side of the bed. After manually adjustments to the probe z values, the first layer is great. I think there is something that is throwing off the sensor on the bed (although it works very reliably because I get the same results over and over again), because you see the left side too close to the nozzle complaint all the time.

I'm hoping to have my new heaterblock soon that will be a direct fit, no drilling out holes for the temp sensor. I have one of those blocks too btw, thermistor won't fit :/

**Comment by BeauSlim** [comment]
I'm also waiting on longer screws before messing more with a copper CR10 block, but I can confirm that a "Generic 3950" glass bead thermistor works fine, at least with Klipper.  I crimped a JST 1.5 plug to a spare I had for another printer.  

@bassamanator I discovered that my X axis is so rigid that the right side can be lifted off its Z screw when the extruder is to the left.  As the weight of the extruder moves to the right, the nozzle (and the probe) get lower, making the path curve down.  You might have the same thing.  Anti-backlash Z nuts (or just tighter nuts) may fix this but I haven't confirmed yet.

**Comment by BeauSlim** [comment]
@bassamanator Sorry for the confusion.  Try this:

With the hotend all the way over to the left, remove the 2 bolts holding the right POM nut and spin the right motor so the black nut moves up an inch or so.  Notice that there is some play between the POM nut and the screw it rides on.  Some play is is normal, as the weight of the gantry is supposed to hold the nuts to the bottom thread surface of the lead screw.  
  
You may also see that the X gantry doesn't droop down on the right side even though nothing is holding it up.  This force is what is lifting the right POM nut off of the thread it is supposed to ride.  And as the hotend moves to the right, the right side of the gantry slowly moves down due to the added weight, creating a curved path for the probe.

**Comment by bassamanator** [comment]
@BeauSlim Thank you for this. Could you please copy and paste your last post to this [thread](https://github.com/bassamanator/everything-sovol-sv06/discussions/22)? I've been discussing this with 2 other people and this issue needs its own thread.
